### PR TITLE
minimal new location creation

### PR DESCRIPTION
**Pull request by esoergel**
Only show required fields on new location creation.  Also a couple minor bugfixes that creeped in.
This has been on staging for a while, although not the latest (smaller) changes.
@benrudolph

### CODE REVIEW

**Comment by czue** [comment]
@esoergel are you explicitly having QA on staging done or are you just putting it on? I don't think locations get accessed on staging very frequently so not sure how good a test that would be

**Comment by czue** [comment]
(didn't see anything blocking this from merge but didn't review closely enough to merge myself)

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -89,6 +90,33 @@ def __init__(self, location, bound_data=None, is_new=False,
...
+        self.helper.layout = crispy.Layout(
+            crispy.Fieldset(*self.get_fields(is_new))
+        )
+
+    def get_fields(self, is_new):
+        if is_new:
+            parent = (Location.get(self.location.parent_id)
+                      if self.location.parent_id else None)
+            child_types = allowed_child_types(self.location.domain, parent)
+            return filter(None, [
```
Comment (benrudolph) [inline corehq/apps/locations/forms.py]: that's a sweet trick

**Comment by benrudolph** [comment]
:+1:

**Comment by benrudolph** [comment]
let me know what you want to do about QA

File: corehq/apps/locations/forms.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/locations/forms.py]: Yeah, I like that one a lot.  It filters anything falsy, not just `None`.

**Comment by esoergel** [comment]
(bug on staging, introduced by new changes on master, I'll update when it's fixed)

**Comment by wpride** [comment]
@esoergel @emord please see merge conflicts I resolved

**Comment by esoergel** [comment]
Sheel took a pass at this, should be good to go.

**Comment by czue** [comment]
:+1:

**Pull request merged**